### PR TITLE
Update to v1.21.3

### PR DESCRIPTION
**Pull request by danpetry**
# Links

[Upstream](https://github.com/Blosc/c-blosc/tree/v1.21.3)
[Diff](https://github.com/Blosc/c-blosc/compare/v1.21.1...v1.21.3)
[Changelog](https://github.com/Blosc/c-blosc/blob/v1.21.3/RELEASE_NOTES.rst)
[Jira ticket](https://anaconda.atlassian.net/browse/PKG-940?atlOrigin=eyJpIjoiMmRjMjM4N2EzZDkwNDlmZmI5YjJmZWZkNmQ5NzNiZjUiLCJwIjoiaiJ9)
[License](https://github.com/Blosc/c-blosc/blob/v1.21.3/LICENSES/BLOSC.txt)

# Changes

- Remove legacy CMake config variables
- Bump version, update SHA256, reset build number
- Include some informative comments
- Pin host name packages to versions used when statically linking upstream, because lower versions might not work and higher versions would reduce compatibility. Zlib's patch version had to be bumped up by one to satisfy zstd's dependency requirements on it.
- Updated about section with additional info

# Further notes

- CMake configuration variables introduced upstream implement previous build options by default.
- Upstream license is stated simply as "BSD", which isn't a license in the [SPDX list](https://spdx.org/licenses/), so I've gone with BSD-3-Clause as previously included and indicated in various upstream locations such as [here](https://github.com/Blosc/c-blosc/blob/ac67275455b8c10938964dba2f584b8f3f12a48d/blosc/blosclz.h)

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -21,10 +21,13 @@ requirements:
...
-    - zlib
-    - lz4-c                                          # [unix]
-    - zstd
+    # we probably won't ever use vc<14 at this point, but this is left here just in case
+    - msinttypes =r26                                # [win and vc<14]
+    # pinnings below are taken from upstream's internal-complibs folder.
+    # zlib's patch version had to be bumped up by one to satisfy zstd's dependency requirements on it.
+    - zlib =1.2.12
+    - lz4-c =1.9.4
+    - zstd =1.5.2
```
Comment (skupr-anaconda) [inline recipe/meta.yaml:30]: `zstd` is already **1.5.2** in `cbc.yaml`, see https://github.com/AnacondaRecipes/aggregate/blob/331d0d2487f7d74cc929e1f0a630a5fa26498e27/conda_build_config.yaml#L230-L231. So just `zstd` will be enough

**Review by varlackc — COMMENTED** [review]
As soon as the changes on the pinnings in the recipe take place, we will be ready to approve.

**Comment by danpetry** [comment]
As per discussion yesterday, pinnings should remain until CI is fixed. I've added a comment though. @varlackc @skupr-anaconda are you happy with that and if so could you approve?

File: recipe/meta.yaml (hunk 1 above)
Comment (danpetry) [reply recipe/meta.yaml]: thanks, you're right, see note below though

**Review by skupr-anaconda — APPROVED** [review]

**Review by varlackc — APPROVED** [review]
LGTM

**Pull request merged**